src/standard_wrappers/nft_standards/dip_721.ts

```diff
@@ -32,77 +49,152 @@ export default class ERC721 extends NFT {
     super(canisterId, agent);
 
     this.actor = Actor.createActor(IDL, {
-      agent, canisterId,
+      agent,
+      canisterId,
     });
   }
 
   async getUserTokens(principal: Principal): Promise<NFTDetails[]> {
-    const userTokensResult = await this.actor.getMetadataForUserDip721(principal);
-    const tokens = userTokensResult || [];
+    const userTokensResult = await (async () => {
```

Comment:
why declare it as an inline function here instead of the plain code?

Reply:
It's for the contextual encapsulation as I just care about the end result for "userTokensResult", not an inline function but an immediately invoked function. The snippet can be moved as a utility if you don't find it easy to read.